### ISSUE TITLE
META: API Discussion

### ISSUE THREAD

**Issue by pzuraq** [kybishop/ember-popper#6]
We're going to be using Popper.js for our app, but since we're on a legacy version of Ember at the moment we can't use this addon (and probably won't be able to use it until that Glimmer `render-in-place` helper arrives). However, we would like to match the external API of this component with our internal implementation so that when that time comes, we can swap it out and start contributing back.

The current API is pretty simple and not really easy to use with pre-Ember 2.4(?) apps - basically any without the `hash` helper, because `options` is difficult to specify declaratively otherwise. We can't even really polyfill the hash helper because in 1.8/1.11, single word helpers are a bit of a pain 😕 

I don't think it would be too unreasonable to follow `ember-tether`s pattern of adding the top level options as properties of the component, but requiring them to be passed in as objects below that. So a configuration currently like so:

```
{{#ember-popper options=(hash placement="top" modifiers=(hash offset=(hash offset="10px")))}}
```

Would become:

```
{{#ember-popper placement="top" modifiers=(hash offset=(hash offset="10px"))}}
```

There are only 3 top level options for Popper, so this would be a pretty minimal API to maintain. If Popper decides to change its public API dramatically, we could follow suite with a major/minor version bump. I also think this would be a lot better for developer ergonomics - the bottom example is much easier to read and understand.

**Comment by pzuraq** [comment]
I also think we should try to preserve `class` instead of `popperClass` as an external API, can do some work to make that happen cleanly

**Comment by kybishop** [comment]
Re flattening: I'm totally on board with this. Will happily accept PRs, or get around to it myself eventually.

Re preserving class: I'd love that. Went asking about this in #ember-cli and was recommended against it without employing a lot of trickery, but if you know a way to do it cleanly, that would be awesome.

**Comment by kybishop** [comment]
Resolved in #11. Will make a separate issue for using `class` over `popperClass`.
